File: TEQClient/teqclient/src/js/api/microservices/pl-similarity.js
import { sendAsync } from "@/js/utility/request.js";
import { GATEWAY } from "@/js/api/microservices/gateway.js";
import { getRequest } from "@/js/api/api.js";

const API = `${GATEWAY}/pl/similarity/`;

export const findScriptSimilarity = (code1, code2) => {
  const request = getRequest("POST", { code1, code2 });

  return sendAsync(API, request);
};

File: TEQServer/mainapp/api/pl_similarity.py
import requests

from mainapp.api.gateway import GATEWAY

URL = f"{GATEWAY}/pl/similarity/"

def calculate_similarity(code1, code2):
    response = requests.post(URL, json={"code1": code1, "code2": code2})

    if response.ok:
        data = response.json()
        if "similarity" not in data:
            raise Exception("No similarity found")
        return data["similarity"]

    raise Exception(f"{response.status_code} - {response.text}")